data/modules/PolicePatrol/PolicePatrol.lua

```diff
@@ -78,50 +85,37 @@ local onShipHit = function (ship, attacker)
 	-- Support all police, not only the patrol
 	local police = ShipDef[Game.system.faction.policeShip]
 	if ship.shipId == police.id and attacker.shipId ~= police.id then
+		piracy = true
 		attackShip(attacker)
 	end
 end
 
 local onShipFiring = function (ship)
-	if piracy then return end
+	if piracy or target then return end
 
 	local police = ShipDef[Game.system.faction.policeShip]
 	if ship.shipId ~= police.id then
 		for i = 1, #patrol do
-			if ship:DistanceTo(patrol[i]) <= 4000000 then
-				Comms.ImportantMessage(string.interp(l_ui_core.X_CANNOT_BE_TOLERATED_HERE, { crime = l_ui_core.PIRACY }, patrol[i].label))
+			if ship:DistanceTo(patrol[i]) <= lawEnforcedRange then
+				Comms.ImportantMessage(string.interp(l_ui_core.X_CANNOT_BE_TOLERATED_HERE, { crime = l_ui_core.UNLAWFUL_WEAPONS_DISCHARGE }, patrol[i].label))
```

Comment:
Note for future reference that `string.interp(l.SOME_STRING, { crime = ... })` is also available via a custom operator as `l.SOME_STRING % { crime = ... }`. You don't need to amend this PR to prefer the use of either one however.